packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -122,8 +142,21 @@ export class LibraryDeploySourcePathExecutor extends DeployRetrieveLibraryExecut
       );
       channelService.showCommandWithTimestamp(`Starting ${this.executionName}`);
 
+      let components: SourceComponent[] = [];
       const registryAccess = new RegistryAccess();
-      const components = registryAccess.getComponentsFromPath(response.data);
+      if (typeof response.data === 'string') {
+        components = registryAccess.getComponentsFromPath(response.data);
+      } else {
+        const allComponents: SourceComponent[] = [];
+
+        for (const filepath of response.data) {
+          allComponents.push(...registryAccess.getComponentsFromPath(filepath));
+        }
+        components = allComponents.filter(
+          (cmp, index) => allComponents.indexOf(cmp) === index
+        );
```

Comment:
re: earlier convo with bryan
we can look into including this logic at the registry level instead -> you can pass in multiple paths to the registry and it will create a unique list of all the components for you

Reply:
Agreed, we should open a bug/work item to add this to the library

Reply:
```typescript
components = allComponents.filter(
    (cmp, index) => allComponents.indexOf(cmp) === index
);
```

I'm not sure I'm following how this dedupes the components. Don't we need to make some unique key based off of the type and fullname?

Reply:
Yeah good catch, I realized I was testing with a shallow copy of an existing element from the array. I'll update this filter to specifically compare the type & full name of each object

Reply:
Could we use a better indexing mechanism? This has O(n**2) complexity, where n is number of components

Reply:
updated!